src/scripts/upload-wpt-results.py

```diff
@@ -106,9 +112,9 @@ def consolidate(raw_results_files, platform_info):
     # this metadata in reports generated via the Sauce Labs service.)
     assert 'run_info' in report
     assert isinstance(data['run_info'], object)
-    for name in (u'product', 'browser_version', 'os', 'os_version'):
-        if name not in report['run_info']:
-            report['run_info'][name] = platform_info[name]
+    if platform_override is not None:
+        for name in platform_override:
+            report['run_info'][name] = platform_override[name]
```

Comment:
Nit: you could drop `is not None`. Empty dict/lists are safe to be skipped, too.

And perhaps simply `report['run_info'].update(platform_override)`

Reply:
Much better!